code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm

```diff
@@ -336,76 +336,136 @@
-			NewTerrainFloors = /turf/open/floor/plating/abductor
-			NewTerrainWalls = /turf/closed/wall/mineral/abductor
-			NewTerrainChairs = /obj/structure/bed/abductor //ayys apparently don't have chairs. An entire species of people who only recline.
-			NewTerrainTables = /obj/structure/table/abductor
-
 /obj/machinery/anomalous_crystal/theme_warp/ActivationReaction(mob/user, method)
-	if(..())
-		var/area/A = get_area(src)
-		if(!A.outdoors && !(A in affected_targets))
-			for(var/atom/Stuff in A)
-				if(isturf(Stuff))
-					var/turf/T = Stuff
-					if((isspaceturf(T) || isfloorturf(T)) && NewTerrainFloors)
-						var/turf/open/O = T.ChangeTurf(NewTerrainFloors, flags = CHANGETURF_IGNORE_AIR)
-						if(prob(florachance) && NewFlora.len && !O.is_blocked_turf(TRUE))
-							var/atom/Picked = pick(NewFlora)
-							new Picked(O)
-						continue
-					if(iswallturf(T) && NewTerrainWalls)
-						T.ChangeTurf(NewTerrainWalls)
-						continue
-				if(istype(Stuff, /obj/structure/chair) && NewTerrainChairs)
-					var/obj/structure/chair/Original = Stuff
-					var/obj/structure/chair/C = new NewTerrainChairs(Original.loc)
-					C.setDir(Original.dir)
-					qdel(Stuff)
-					continue
-				if(istype(Stuff, /obj/structure/table) && NewTerrainTables)
-					new NewTerrainTables(Stuff.loc)
-					continue
-			affected_targets += A
+	. = ..()
+	if (!.)
+		return FALSE
+	var/area/current_area = get_area(src)
+	if (current_area in affected_targets)
+		return FALSE
+	if (terrain_theme.transform_area(current_area))
+		affected_targets += current_area
+	return TRUE
+
+/// Transforms an area's turfs and objects into a different theme
+/datum/crystal_warp_theme
+	/// Friendly name of theme
+	var/name = ""
+	/// Typepath of floor to replace open turfs with
+	var/floor
+	/// Typepath of wall to replace closed turfs with
+	var/wall
+	/// Typepath of object to replace chairs with
+	var/chair
+	/// Typepath of object to replace tables with
+	var/table
+	/// Typepath of things to potentially spawn on transformed open turfs
+	var/list/flora_and_fauna = list()
+	/// Chance per turf to create the things in the list above
+	var/flora_and_fauna_chance = 8
+
+/// Change appropriate objects in provided area to those matching our theme, and spawn some plants or animals
+/datum/crystal_warp_theme/proc/transform_area(area/target_area)
+	if (target_area.outdoors)
+		return FALSE
+	for(var/atom/thing in target_area)
+		if(isturf(thing))
+			replace_turf(thing)
+			continue
+		if(chair && istype(thing, /obj/structure/chair))
+			replace_object(thing, chair)
+			continue
+		if(table && istype(thing, /obj/structure/table))
+			replace_object(thing, table)
+			continue
+	return TRUE
+
+/// Replaces a turf with a different themed turf
+/datum/crystal_warp_theme/proc/replace_turf(turf/target_turf)
+	if(isindestructiblefloor(target_turf) || isindestructiblewall(target_turf) || isopenspaceturf(target_turf))
+		return
+
+	if(iswallturf(target_turf) && wall)
```

Comment:
```suggestion
	if(wall && iswallturf(target_turf))
```

standardize it with the rest of your code (i'm going to do this myself since it's such a nitpicky nitpick)